### ISSUE TITLE
Removing video leaves screen white?

### ISSUE THREAD

**Issue by wootwoot1234** [TheWidlarzGroup/react-native-video#58]
I'm new to react native so I'm probably just doing something wrong...

For some reason, when I'm done playing the video the Video (element?) stays but with no video.  

Basically, after a given amount of time, I from the onProgress callback I call `this.props.closeVideo();` in the child which calls `closeVideo()` in the parent and sets `showReward` to false then the screen goes white.

Here's my code.
## Parent

```
handleTouchStart(event: Object) {
    this.setState({showReward: true});
}
closeVideo() {
    console.log("closeVideo()");
    this.setState({showReward: false});
}
render() {
    if(this.state.showReward) {
        return (
            <Reward closeVideo={() => this.closeVideo()} />
            );
    } else {
        return (
            <View
                style={styles.container}
                onTouchStart={(event) => this.handleTouchStart(event)}>
                <Tile state={this.state}/>
            </View>
        );
    }
}
```
## Reward (child)

```
setTime(progress) {
    if(progress.currentTime > this.state.reinforcerLength) {
        this.props.closeVideo();
    }
}
render(){
    return (
        <Video source={{uri: "SampleVideo"}} // Can be a URL or a local file.
            resizeMode="cover"           // Fill the whole screen at aspect ratio.
            repeat={true}                // Repeat forever.
            onProgress={this.setTime.bind(this)}    // Callback every ~250ms with currentTime
            style={styles.backgroundVideo} />
    );
}
```

**Comment by brentvatne** [comment]
Just to rule out the obvious, does the `else` condition in the parent ever render? If you remove the if entirely and just render that, does it work?

Could be a bug if that works.

**Comment by wootwoot1234** [comment]
It was working before I added the video code so yes. :(

I can double check when I get back to my computer in an hour. 

(null)

**Comment by brentvatne** [comment]
@wootwoot1234 - great thanks. If it is working, then if you can create an example repo where it is reproduced that would go a long way to helping me fix it!

**Comment by wootwoot1234** [comment]
@brentvatne  Thanks for your help.

Here's a link to a repository with the bug. 

https://github.com/wootwoot1234/VideoBugExample

Let me know if there is anything else I can do.

**Comment by wootwoot1234** [comment]
Any updates progress on this?

**Comment by brentvatne** [comment]
@wootwoot1234 - thank you so much for the example! That helps a lot. Confirmed that this is fixed in 0.4.6 - I pulled the project, ran it against 0.4.5, it was broken then updated to 0.4.6 and it works! This commit fixes it: a435efe3d0aeed3267165487c2d708b7d8fe0a77

**Comment by wootwoot1234** [comment]
@brentvatne just updated it and it works great.  Thanks!
